src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -214,6 +214,136 @@ internal static string ColumnEncryptionSettingToString(SqlConnectionColumnEncryp
             }
         }
 
+        #region <<AttestationProtocol Utility>>
+
+        /// <summary>
+        /// Attestation Protocol.
+        /// </summary>
+        const string AttestationProtocolHGS = "HGS";
+        const string AttestationProtocolAAS = "AAS";
+
+        /// <summary>
+        ///  Convert a string value to the corresponding SqlConnectionAttestationProtocol
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="result"></param>
+        /// <returns></returns>
+        internal static bool TryConvertToAttestationProtocol(string value, out SqlConnectionAttestationProtocol result)
+        {
+            bool isSuccess = false;
+
+            if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolHGS))
+            {
+                result = SqlConnectionAttestationProtocol.HGS;
+                isSuccess = true;
+            }
+            else if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolAAS))
+            {
+                result = SqlConnectionAttestationProtocol.AAS;
+                isSuccess = true;
+            }
+            else
+            {
+                result = DbConnectionStringDefaults.AttestationProtocol;
+            }
+
+            return isSuccess;
```

Comment:
Can be simplified and get rid of the isSuccess variable 
```
            if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolHGS))
            {
                result = SqlConnectionAttestationProtocol.HGS;
                return true;
            }
            else if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolAAS))
            {
                result = SqlConnectionAttestationProtocol.AAS;
                return true;
            }
            return false;
```

Reply:
Changed